pkg/collector/corechecks/servicediscovery/module/impl_linux.go

```diff
@@ -686,22 +687,18 @@ func (s *discovery) enrichContainerData(service *model.Service, containers map[s
 	serviceName := getServiceNameFromContainerTags(container.Tags)
 
 	if serviceName != "" {
-		service.GeneratedName = serviceName
+		service.ContainerServiceName = serviceName
 		// Update the legacy name field as well
-		if service.DDService == "" {
+		if service.Name == "" {
```

Comment:
Is this code for setting `service.Name` needed now? Maybe we should just leave that to be ddservice/generated and not involve the container name there?